tests/components/azure_event_hub/test_init.py

```diff
@@ -90,128 +37,166 @@ async def test_full_config(hass):
             },
         }
     }
-    config[azure_event_hub.DOMAIN].update(MIN_CONFIG)
+    config[azure_event_hub.DOMAIN].update(CS_CONFIG_FULL)
     assert await async_setup_component(hass, azure_event_hub.DOMAIN, config)
 
 
-async def _setup(hass, mock_call_later, filter_config):
-    """Shared set up for filtering tests."""
-    config = {azure_event_hub.DOMAIN: {"filter": filter_config}}
-    config[azure_event_hub.DOMAIN].update(MIN_CONFIG)
-
+async def test_filter_only_config(hass):
+    """Test the popping of the filter and further import of the config."""
+    config = {
+        azure_event_hub.DOMAIN: {
+            "filter": {
+                "include_domains": ["light"],
+                "include_entity_globs": ["sensor.included_*"],
+                "include_entities": ["binary_sensor.included"],
+                "exclude_domains": ["light"],
+                "exclude_entity_globs": ["sensor.excluded_*"],
+                "exclude_entities": ["binary_sensor.excluded"],
+            },
+        }
+    }
     assert await async_setup_component(hass, azure_event_hub.DOMAIN, config)
-    await hass.async_block_till_done()
-    mock_call_later.assert_called_once()
-    return mock_call_later.call_args[0][2]
 
 
-async def _run_filter_tests(hass, tests, process_queue, mock_batch):
-    """Run a series of filter tests on azure event hub."""
+async def test_unload_entry(hass, entry):
+    """Test being able to unload an entry."""
+    assert await hass.config_entries.async_unload(entry.entry_id)
+    assert entry.state == ConfigEntryState.NOT_LOADED
+
+
+async def test_failed_test_connection(hass, mock_get_eventhub_properties):
+    """Test being able to unload an entry."""
+    entry = MockConfigEntry(
+        domain=azure_event_hub.DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    mock_get_eventhub_properties.side_effect = EventHubError("Test")
+    try:
+        await hass.config_entries.async_setup(entry.entry_id)
+    except azure_event_hub.ConfigEntryNotReady:
```

Comment:
We can remove this try... except. This exception is handled by the config entries system.